### PR TITLE
Support for code list management web services

### PR DESCRIPTION
**Pull request by teodoradobreva**

### CODE REVIEW

File: src/04 - Web services.md [hunk 1]
```diff
@@ -56,6 +58,160 @@ Note that all services must remain backwards-compatible even when their version
...
+- **code**
+  - code or short name of the code list value to create or update
+  - unique for the particular code list
+- **name**
+  - user-friendly name of the code list value
+- **description**
+  - description of the code list value
+- **authority** (optional)
+  - authority of the code list value
+  - applicable only for the *skjerming* code list
```
[the remark this answers is not in the document]
Comment (teodoradobreva) [reply src/04 - Web services.md]: Changed.

File: src/02 - Object types.md [hunk 2]
```diff
@@ -154,81 +154,81 @@ The following notation is used in the tables below:
...
+| M021 | string          | beskrivelse                   |          |             |                |  |  x   |  x   |   x   |  x   |               |                                                                                                                      |                                                                                   |
+| M300 | string          | dokumentmedium                |    x     |      x      | dokumentmedium |  |  x   |  x   |   x   |  x   |               |                                                                                                                      |                                                                                   |
+| M500 | string          | skjerming                     |          |             | skjerming      |  |  x   |  x   |   x   |  x   |               | <p>&bull; code list in Documaster</p>                                                                                |                                                                                   |
```
[the remark this answers is not in the document]
Comment (douzounov) [reply src/02 - Object types.md]: Let's keep it for now. We'll remove it in a future version of this documentation.

File: src/04 - Web services.md [hunk 3]
```diff
@@ -56,6 +58,160 @@ Note that all services must remain backwards-compatible even when their version
...
+  - **field**
+    - field in the object type that this code list applies to
+    - **code**
+      - code or short name of the code list value
+    - **name**
+      - user-friendly name of the code list value
+    - **description**
+      - description of the code list value
+    - **authority** (optional)
+      - authority of the code list value
```
Comment (douzounov) [inline src/04 - Web services.md]: Maybe we should add `applicable only to the *skjerming* code list` here.

File: src/04 - Web services.md [hunk 4]
```diff
@@ -56,6 +58,160 @@ Note that all services must remain backwards-compatible even when their version
...
+###### Details
+
+- **results**
+  - array of one or more code lists
+  - **type**
+    - object type that this code list applies to
+  - **field**
+    - field in the object type that this code list applies to
+    - **code**
+      - code or short name of the code list value
```
Comment (douzounov) [inline src/04 - Web services.md]: How about `code (identifier of the code list value)` or `code which uniquely identifies the code list value` or something along these lines?

File: src/04 - Web services.md [hunk 5]
```diff
@@ -56,6 +58,160 @@ Note that all services must remain backwards-compatible even when their version
...
+}
+```
+
+###### Details
+
+- **listId**
+  - ID of the code list in which to create or update the code list value
+  - corresponds to a code list as described in - [03 - Code lists](03%20-%20Code%20lists.md)
+- **code**
+  - code or short name of the code list value to create or update
```
Comment (douzounov) [inline src/04 - Web services.md]: See my other comment about describing the code as a unique identifier of the code list value.

File: src/04 - Web services.md [hunk 6]
```diff
@@ -56,6 +58,160 @@ Note that all services must remain backwards-compatible even when their version
...
+  "name": string,
+  "description": string,
+  "authority": string
+}
+```
+
+###### Details
+
+- **code**
+  - code or short name of the created or updated code list value
```
Comment (douzounov) [inline src/04 - Web services.md]: See previous comments about how to refer to the code.

File: src/04 - Web services.md [hunk 7]
```diff
@@ -56,6 +58,160 @@ Note that all services must remain backwards-compatible even when their version
...
+- **name**
+  - user-friendly name of the code list value
+- **description**
+  - description of the code list value
+- **authority** (optional)
```
Comment (douzounov) [inline src/04 - Web services.md]: Wondering if we could simply say in this document that all code lists have the same base fields, but some lists have extra fields, and not document `authority` here. Now we are marking `authority` as optional, but when reading the documentation, I get the feeling that I can set it on every code list.

File: src/04 - Web services.md [hunk 8]
```diff
@@ -56,6 +58,160 @@ Note that all services must remain backwards-compatible even when their version
...
+Authorization: Bearer ACCESS_TOKEN
+```
+
+###### Details
+
+- **listId**
+  - ID of the code list from which to delete the code list value
+  - corresponds to a code list as described in - [03 - Code lists](03%20-%20Code%20lists.md)
+- **code**
+  - code or short name of the code list value to delete
```
Comment (douzounov) [inline src/04 - Web services.md]: See other comments about how to refer to `code`.

File: src/04 - Web services.md (hunk 4 above)
Comment (teodoradobreva) [reply src/04 - Web services.md]: Changed to `unique identifier of the code list value in the particular code list`.

File: src/04 - Web services.md (hunk 7 above)
Comment (teodoradobreva) [reply src/04 - Web services.md]: Removed `authority` from requests and responses, description is added at the beginning of the code lists section.

File: src/04 - Web services.md [hunk 9]
```diff
@@ -56,6 +58,157 @@ Note that all services must remain backwards-compatible even when their version
...
 - Dates are returned by the web services as **ISO 8601** formatted strings.
 
+# Code list management web services
+
+Address:
+```
+https://{server}:{port}/rms/api/public/noark5/v1/code-lists
+```
+
+Note that the values of all code lists have the same base fields (*code*, *name*, *description*), but some values have extra fields.
```
Comment (ivaylomitrev) [inline src/04 - Web services.md]: I like this approach much better too. I would just suggest a slight rewording (if you agree with it):
```
Special note: All code list values have the same base fields: *code*, *name*, and *description*. Certain code list values, however, also support additional fields. They are described below.

The *skjerming* ...
```

Comment (teodoradobreva) [reply src/04 - Web services.md]: Changed.

**Comment by ivaylomitrev** [comment]
Rebased and squashed in preparation for merge.

**Pull request merged**